app/controllers/application_controller.rb

```diff
@@ -1,3 +1,12 @@
 class ApplicationController < ActionController::Base
+  before_action :set_locale
 
+  def set_locale
+    I18n.locale = params[:locale] if params[:locale].present?
```

Comment:
Was ist mit Settings eines Benutzers? Vielleicht hat schon ein Benutzer bei sich die Sprache eingestellt und will das nicht immer neu machen. Also, vielleicht die Einstellung wenigstens in Cookies speichern und von da auslesen.